tools/utils.py

```diff
@@ -35,3 +35,17 @@ def check_docker_status():
       'is_installed': True,
       'is_running': False
     }
+
+"""
+this method continually prompts the user to enter a "valid" input. 
+validator should be a one argument function which returns a Boolean 
+(whether the input is valid or invalid)
+Returns:
+    the valid user input
+"""
+def prompt_user(prompt, validator):
+    ans = input(prompt)
```

Comment:
this is a good function to have, should we also abstract the `(y or n)` to this as well ?

Reply:
good idea!